libsolidity/interface/StandardCompiler.cpp

```diff
@@ -623,8 +623,135 @@ std::variant<OptimiserSettings, Json::Value> parseOptimizerSettings(Json::Value
 	return { std::move(settings) };
 }
 
+/// Only parses the input settings and fills the InputsAndSettings with it.
+/// If this fails, it does return a JSON error response, otherwise nullopt on success.
+optional<Json::Value> parseInputSettings(Json::Value _settings, StandardCompiler::InputsAndSettings& _output)
+{
+	if (auto result = checkSettingsKeys(_settings))
+		return *result;
+
+	if (_settings.isMember("stopAfter"))
+	{
+		if (!_settings["stopAfter"].isString())
+			return formatFatalError("JSONError", "\"settings.stopAfter\" must be a string.");
+
+		if (_settings["stopAfter"].asString() != "parsing")
+			return formatFatalError("JSONError", "Invalid value for \"settings.stopAfter\". Only valid value is \"parsing\".");
+
+		_output.stopAfter = CompilerStack::State::Parsed;
+	}
+
+	if (_settings.isMember("parserErrorRecovery"))
+	{
+		if (!_settings["parserErrorRecovery"].isBool())
+			return formatFatalError("JSONError", "\"settings.parserErrorRecovery\" must be a Boolean.");
+		_output.parserErrorRecovery = _settings["parserErrorRecovery"].asBool();
+	}
+
+	if (_settings.isMember("viaIR"))
+	{
+		if (!_settings["viaIR"].isBool())
+			return formatFatalError("JSONError", "\"settings.viaIR\" must be a Boolean.");
+		_output.viaIR = _settings["viaIR"].asBool();
+	}
+
+	if (_settings.isMember("evmVersion"))
+	{
+		if (!_settings["evmVersion"].isString())
+			return formatFatalError("JSONError", "evmVersion must be a string.");
+		std::optional<langutil::EVMVersion> version = langutil::EVMVersion::fromString(_settings["evmVersion"].asString());
+		if (!version)
+			return formatFatalError("JSONError", "Invalid EVM version requested.");
+		_output.evmVersion = *version;
+	}
+
+	if (_settings.isMember("debug"))
+	{
+		if (auto result = checkKeys(_settings["debug"], {"revertStrings"}, "settings.debug"))
+			return *result;
```

Comment:
This part needs to be redone. It's out of date. In the current form it reverts everything related to debug info that was added in #11944. Note for example that in this diff `{"revertStrings", "debugInfo"}` gets reverted back to `{"revertStrings"}`. Same with the whole `if (settings["debug"].isMember("debugInfo"))` condition that should be just below.